docs/architecture/adr-053-state-sync-prototype.md

```diff
@@ -72,41 +71,45 @@ message ResponseListSnapshots {
 // Offers a snapshot to the application
 message RequestOfferSnapshot {
     Snapshot snapshot = 1;
-    bytes app_hash = 2;
+    bytes    app_hash = 2;
 }
 
 message ResponseOfferSnapshot {
-    bool accepted = 1;
-    Reason reason = 2;        // Reason why snapshot was rejected
-    enum Reason {
-        unknown = 0;          // Unknown or generic reason
-        invalid_height = 1;   // Height is rejected: avoid this height
-        invalid_format = 2;   // Format is rejected: avoid this format
+    bool   accepted = 1;
+    Reason reason = 2;
+
+    enum Reason {            // Reason why snapshot was rejected
+        unknown        = 0;  // Unknown or generic reason
+        invalid_height = 1;  // Height is rejected: avoid this height
+        invalid_format = 2;  // Format is rejected: avoid this format
+        internal_error = 3;  // Something unexpected went wrong
     }
 }
 
-// Fetches a snapshot chunk
-message RequestGetSnapshotChunk {
+// Loads a snapshot chunk
+message RequestLoadSnapshotChunk {
     uint64 height = 1;
     uint32 format = 2;
-    uint32 chunk = 3;
+    uint32 chunk  = 3; // Zero-indexed
 }
 
-message ResponseGetSnapshotChunk {
-    SnapshotChunk chunk = 1;
+message ResponseLoadSnapshotChunk {
+    bytes chunk = 1;
 }
 
 // Applies a snapshot chunk
 message RequestApplySnapshotChunk {
-    SnapshotChunk chunk = 1;
+    bytes chunk = 1;
 }
 
 message ResponseApplySnapshotChunk {
-    bool applied = 1;
-    Reason reason = 2;      // Reason why chunk failed
-    enum Reason {
-        unknown = 0;        // Unknown or generic reason
-        verify_failed = 1;  // Chunk verification failed
+    bool   applied = 1;
+    Reason reason  = 2;      // Reason why chunk failed
+
+    enum Reason {            // Reason why chunk failed
+        unknown        = 0;  // Unknown or generic reason
+        verify_failed  = 1;  // Snapshot verification failed
+        internal_error = 2;  // Something unexpected went wrong
```

Comment:
isn't this the same as 0 (unknown)?

Reply:
The idea was that `unknown` would be "rejected for an unknown reason", implying that Tendermint should offer other snapshots instead, while `internal_error` would cause Tendermint to abort. However, I suppose we could continue trying other snapshots for internal errors too, in which case we can remove this.